tests/cryolo_relion_it/test_icebreaker_external_job.py

```diff
@@ -0,0 +1,71 @@
+from relion.cryolo_relion_it import icebreaker_histogram
+import os
+import gemmi
+
+
+def test_create_json_histogram(tmpdir):
+    # Prep
+    os.chdir(tmpdir)
+    external_dir = tmpdir.mkdir("External")
+    icebreaker_dir = external_dir.mkdir("Icebreaker_group")
+    particles_file = icebreaker_dir.join("particles.star")
+    particles_file.write("")
+
+    assert icebreaker_histogram.create_json_histogram(tmpdir) == "ice_hist.json"
```

Comment:
```suggestion
def test_create_json_histogram(tmp_path):
    icebreaker_dir = tmp_path / "External" / "Icebreaker_group"
    icebreaker_dir.mkdir(parents=True)
    particles_file = icebreaker_dir / "particles.star"
    particles_file.write_text("")

    assert icebreaker_histogram.create_json_histogram(tmp_path) == "ice_hist.json"
    assert tmp_path.joinpath("ice_hist.json").is_file()
```
I really would like to avoid `chdir()` if at all possible. The reasoning here is that this affects what is essentially a process-wide singleton, and this has effects beyond the lifetime of this particular one test. Worst case this can lead to interesting effects where the outcome of another test depends on what tests you ran previously, and in what order.

Within the main code using `chdir()` is obviously even worse, as the execution paths are less well determined. On the other hand, if you can make code independent of the current working directory then, if it is used as a library, it will instantly become be more generally useful.

Secondly, we should no longer use `tmpdir` in new tests. Use `tmp_path` instead. The former returns a different path object type that is no longer actively developed, the latter returns a standard python pathlib object. Their attributes, conventions, and methods are slightly different, and it's much easier to only use the standard one rather than two different ones.

Finally, added a test to make sure the `.json` file exists.

(These suggestions will also apply to the remaining tests to varying degrees.)

Reply:
Agree with these changes, with the exception of the added test. 
I don't think this will pass, because of the returning None if there is no data in the file. `test_create_json_histogram_creates_a_file` writes some data to an input file and then the output file is created, which is why I currently have the file check in there rather than in `test_create_json_histogram`.

Reply:
👍